File: app/src/main/java/com/soumya/wwdablu/zomatobuddy/common/DefaultCuisineImage.java
package com.soumya.wwdablu.zomatobuddy.common;

import android.support.annotation.DrawableRes;

import com.soumya.wwdablu.zomatobuddy.R;

import java.util.HashMap;

public class DefaultCuisineImage {

    private static HashMap<String, Integer> cuisineDefaultCardImage;
    static {
        initCuisineDefaultCardImage();
    }

    public static synchronized @DrawableRes int getCuisineImage(String cuisine) {

        @DrawableRes int drawableRes;

        if(cuisineDefaultCardImage.containsKey(cuisine.toLowerCase())) {
            drawableRes = cuisineDefaultCardImage.get(cuisine.toLowerCase());
        } else {
            drawableRes = cuisineDefaultCardImage.get("default");
        }

        return drawableRes;
    }

    private static void initCuisineDefaultCardImage() {

        if(null == cuisineDefaultCardImage) {
            cuisineDefaultCardImage = new HashMap<>();
        }

        //No need to insert, already its done
        if(0 != cuisineDefaultCardImage.size()) {
            return;
        }

        cuisineDefaultCardImage.put("indian", R.drawable.default_food_indian);
        cuisineDefaultCardImage.put("mexican", R.drawable.default_food_mexican);
        cuisineDefaultCardImage.put("american", R.drawable.default_food_american);
        cuisineDefaultCardImage.put("chinese", R.drawable.default_food_chinese);
        cuisineDefaultCardImage.put("italian", R.drawable.default_food_italian);
        cuisineDefaultCardImage.put("japanese", R.drawable.default_food_japanese);
        cuisineDefaultCardImage.put("default", R.drawable.default_food);
    }
}

File: app/src/main/java/com/soumya/wwdablu/hungry/common/DefaultCuisineImage.kt
package com.soumya.wwdablu.hungry.common

import androidx.annotation.DrawableRes
import com.soumya.wwdablu.hungry.R
import java.util.*

object DefaultCuisineImage {
    private var cuisineDefaultCardImage: HashMap<String?, Int?> = HashMap()

    init {
        initCuisineDefaultCardImage()
    }

    @DrawableRes
    @Synchronized
    fun getCuisineImage(cuisine: String?): Int {
        return if (cuisineDefaultCardImage.containsKey(cuisine?.toLowerCase())) {
            cuisineDefaultCardImage[cuisine?.toLowerCase()]!!
        } else {
            cuisineDefaultCardImage["default"]!!
        }
    }

    private fun initCuisineDefaultCardImage() {

        //No need to insert, already its done
        if (0 != cuisineDefaultCardImage.size) {
            return
        }

        cuisineDefaultCardImage["indian"] = R.drawable.default_food_indian
        cuisineDefaultCardImage["mexican"] = R.drawable.default_food_mexican
        cuisineDefaultCardImage["american"] = R.drawable.default_food_american
        cuisineDefaultCardImage["chinese"] = R.drawable.default_food_chinese
        cuisineDefaultCardImage["italian"] = R.drawable.default_food_italian
        cuisineDefaultCardImage["japanese"] = R.drawable.default_food_japanese
        cuisineDefaultCardImage["default"] = R.drawable.default_food
    }

    init {
        initCuisineDefaultCardImage()
    }
}